rmf_traffic/cmake/rmf_traffic-config.cmake.in

```diff
@@ -4,11 +4,24 @@ get_filename_component(rmf_traffic_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 
 include(CMakeFindDependencyMacro)
 
+find_dependency(eigen3_cmake_module REQUIRED)
 find_dependency(Eigen3 REQUIRED)
 find_dependency(rmf_utils REQUIRED)
 
 if(NOT TARGET rmf_traffic::rmf_traffic)
     include("${rmf_traffic_CMAKE_DIR}/rmf_traffic-targets.cmake")
 endif()
 
+if (NOT eigen3_cmake_module_FOUND)
+  set(rmf_traffic_FOUND 0)
```

Comment:
According to this line in [the documentation](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html)

> If the dependency could not be found it sets an informative diagnostic message and calls return() to end processing of the calling package configuration file

we shouldn't need any of this logic, because `find_dependency(eigen3_cmake_module)` up above should return right away if the module isn't found.

Reply:
True, I wonder if the message thrown is different though :thinking: I was just following the README instructions [here](https://github.com/ros2/eigen3_cmake_module#using-without-ament_cmake) but they use `find_package` and then do the check. 

I have the feeling that the idea there is to be able to set `rmf_traffic_FOUND` to 0 so the message is something like `rmf_traffic not found` which I wonder if it happens when we do `find_dependency`. I'm also not sure if there's any consequences in the case `cmake` is not setting `rmf_traffic_FOUND` and it's just throwing something on the line of `eigen3_cmake_module not found` instead, the only drawback I can think of is that the user might be confused as it will not be a dependency directly written on their `cmake` file.

I guess we either do: `find_package + if` or `find_dependency`, any preference?